lisa/sut_orchestrator/util/device_pool.py

```diff
@@ -0,0 +1,63 @@
+from typing import Any, List, Optional
+
+from lisa.sut_orchestrator.util.schema import HostDevicePoolSchema, HostDevicePoolType
+from lisa.util import LisaException, SkippedException
+
+
+class BaseDevicePoolImpl:
+    def _create_device_pool(
```

Comment:
The prefix `_` means protected or private methods, which means call them only in this class or subclasses. This base class has only private methods, it's not a normal design.

Reply:
@squirrelsc 

Only class or child-subclasses are using that method. Each platform has the device pool implementation which inherit the `BaseDevicePool` for common functionality. `BaseDevicePool` methods are protected so child class can use it and that is why i used `"_"` with function names. None of the method are private, otherwise i would have used `"__"`

We used companion class to reduce the ancestors for child classes like `CloudHypervisorPlatform `and `QemuPlatform`

Reply:
at least the `_configure_device_passthrough_pool` should be public, and move upper than all private methods. A class needs at least one public method, or other class cannot call it at all.

Reply:
@squirrelsc Done. Please check latest push